MDNetworking/MemberImpl/MDReplicatedMember.cs

```diff
@@ -1,69 +1,111 @@
+        protected MethodInfo OnValueChangedCallback = null;
 
         public MDReplicatedMember(MemberInfo Member, bool Reliable, MDReplicatedType ReplicatedType, WeakRef NodeRef,
             MDReplicatedSetting[] Settings)
         {
             MDLog.AddLogCategoryProperties(LOG_CAT, new MDLogProperties(MDLogLevel.Info));
+            GameSession = MDStatics.GetGameSession();
+            Replicator = GameSession.Replicator;
+            GameClock = GameSession.GetGameClock();
+
             this.Member = Member;
             this.Reliable = Reliable;
             this.ReplicatedType = ReplicatedType;
             this.NodeRef = NodeRef;
+            ParseSettings(MDReplicator.ParseParameters(typeof(Settings), Settings));
             CheckIfShouldReplicate();
         }
 
-        ///<summary>Provides a unique key to identify this member</summary>
-        public string GetUniqueKey()
-        {
-            Node Node = NodeRef.GetRef() as Node;
-            return Node?.GetPath() + "#" + Member.Name;
-        }
-
-        protected virtual object GetValue()
+        protected void ParseSettings(MDReplicatedSetting[] SettingsValues)
         {
-            Node Instance = NodeRef.GetRef() as Node;
-            FieldInfo Field = Member as FieldInfo;
+            foreach (MDReplicatedSetting setting in SettingsValues)
+            {
+                switch ((Settings) setting.Key)
+                {
+                    case Settings.OnValueChangedEvent:
+                        Node Node = NodeRef.GetRef() as Node;
+                        OnValueChangedCallback = Node.GetType().GetMethod(setting.Value.ToString(),
+                            BindingFlags.Public | BindingFlags.NonPublic | 
+                            BindingFlags.FlattenHierarchy | BindingFlags.Instance);
+                        break;
+                    case Settings.Converter:
+                        Type DataConverterType = Type.GetType(setting.Value.ToString());
+                        if (DataConverterType != null && DataConverterType.IsAssignableFrom(typeof(IMDDataConverter)))
+                        {
+                            DataConverter = Activator.CreateInstance(DataConverterType) as IMDDataConverter;
+                        }
+                        break;
+                }
+            }
 
-            if (Field != null)
+            // We got no data converter setting
+            if (DataConverter == null)
             {
-                return Field.GetValue(Instance);
+                if (Member.GetUnderlyingType().GetInterface(nameof(IMDDataConverter)) != null)
+                {
+                    DataConverter = Activator.CreateInstance(Member.GetUnderlyingType()) as IMDDataConverter;
+                }
+                else if (Member.GetUnderlyingType().Namespace != "System" && Member.GetUnderlyingType().Namespace != "Godot"
+                && Member.GetUnderlyingType().Namespace.StartsWith("Godot.") == false
```

Comment:
Something to add to the style guide, conditions/function arguments that are split over multiple lines should have the extra lines indented